vlm/match.py

```diff
@@ -0,0 +1,103 @@
+from aiohttp.web import HTTPBadRequest
+import hail as hl
+import os
+
+VLM_DATA_DIR = os.environ.get('VLM_DATA_DIR')
+SEQR_BASE_URL = os.environ.get('SEQR_BASE_URL')
+NODE_ID = os.environ.get('NODE_ID')
+
+BEACON_HANDOVER_TYPE = {
+    'id': NODE_ID,
+    'label': f'{NODE_ID} browser'
+}
+
+BEACON_META = {
+    'apiVersion': 'v1.0',
+    'beaconId': 'com.gnx.beacon.v2',
+    'returnedSchemas': [
+        {
+            'entityType': 'genomicVariant',
+            'schema': 'ga4gh-beacon-variant-v2.0.0'
+        }
+    ]
+}
+
+QUERY_PARAMS = ['assemblyId', 'referenceName', 'start', 'referenceBases', 'alternateBases']
+
+GENOME_VERSION_GRCh38 = 'GRCh38'
+GENOME_VERSION_GRCh37 = 'GRCh37'
+ASSEMBLY_LOOKUP = {
+    GENOME_VERSION_GRCh37: GENOME_VERSION_GRCh37,
+    GENOME_VERSION_GRCh38: GENOME_VERSION_GRCh38,
+    'hg38': GENOME_VERSION_GRCh38,
+    'hg19': GENOME_VERSION_GRCh37,
+}
+
+def get_variant_match(query: dict) -> dict:
+    chrom, pos, ref, alt, genome_build = _parse_match_query(query)
+
+    locus = hl.locus(chrom, pos, reference_genome=genome_build)
+    interval = hl.eval(hl.interval(locus, locus, includes_start=True, includes_end=True))
+    ht = hl.read_table(
+        f'{VLM_DATA_DIR}/{genome_build}/SNV_INDEL/annotations.ht', _intervals=[interval], _filter_intervals=True,
+    )
+    ht = ht.filter(ht.alleles==hl.array([ref, alt]))
+    counts = ht.aggregate(hl.agg.take(ht.gt_stats, 1))
+
+    return _format_results(counts, genome_build, f'{chrom}-{pos}-{ref}-{alt}')
+
+
+def _parse_match_query(query: dict) -> tuple[str, int, str, str, str]:
+    missing_params = [key for key in QUERY_PARAMS if key not in query]
+    if missing_params:
```

Comment:
If there's not a huge rush on getting this in, now might be a good time to explore [pydantic validation](https://docs.pydantic.dev/latest/concepts/validators/#annotated-validators).

Reply:
Theres not a rush to get this specific code out, but I have other things on my plate that I think are higher priority for me. So its less about getting this code released and more about freeing up my time for other feature work